howto/android/custom-vhal.md

```diff
@@ -136,6 +136,100 @@ addons:
+
+       <remote name="aosp"
+          fetch=".."
+          review="https://android-review.googlesource.com/" />
+
+       <remote name="github"
+          fetch="https://github.com/canonical/" />
+
+1. Add a new project named `vendor_canonical_interfaces` to the newly added remote and set it to use the `main` branch.
+
+       <project path="vendor/canonical/interfaces"
+                name="vendor_canonical_interfaces"
+                remote="github"
+                revision="main" />
+
+1. Sync the project with the remote.
+
+        repo sync vendor_canonical_interfaces
+
+1. In the VHAL implementation, create a VHAL manifest fragment named `vendor.canonical.interfaces.vehicle@1.0.xml`.
+
+       <manifest version="1.0" type="device">
+           <hal format="hidl">
+               <name>vendor.canonical.interfaces.vehicle</name>
+               <transport>hwbinder</transport>
+               <fqname>@1.0::IVehicle/default</fqname>
+           </hal>
+       </manifest>
+
+   Place the VHAL manifest fragment file in the same folder as the `Android.bp` file that declares the VHAL service, and include the Anbox VHAL manifest fragment in the VHAL service declaration within the `Android.bp` file. Additionally, add the HIDL module as a shared library that the VHAL service links to in the `Android.bp` file.
+
+
+       cc_binary {
+           name: "vendor.<company>.vehicle@<version>-service",
+           vintf_fragments: [
+               ...
+               "vendor.canonical.interfaces.vehicle@1.0.xml",
+           ],
+           shared_libs: [
+               ...
+               "vendor.canonical.interfaces.vehicle@1.0",
+           ],
+           ...
+       }
+
+1. Take the [example](https://github.com/canonical/vendor_canonical_interfaces/tree/main/vehicle/1.0/default) as a reference, which implements the `IVehicle` interface.
+
+        Return<void> VHalService::get(
+            const VehiclePropValue& requestedPropValue, IVehicle::get_cb _hidl_cb) {
+          uid_t uid = android::IPCThreadState::self()->getCallingUid();
+          if (uid != AID_VEHICLE_NETWORK) {
+            _hidl_cb(StatusCode::ACCESS_DENIED, kEmptyValue);
+            return Void();
+          }
+
+          // NOTE: a VHAL implementation must allow access to non-readable vehicle properties.
+          return Void();
+        }
+
+        Return<StatusCode> VHalService::set(const VehiclePropValue& value) {
+          uid_t uid = android::IPCThreadState::self()->getCallingUid();
+          if (uid != AID_VEHICLE_NETWORK)
+            return StatusCode::ACCESS_DENIED;
+
+          // NOTE: a VHAL implementation must allow modification of non-writable vehicle properties.
+          return StatusCode::NOT_AVAILABLE;
+        }
+
+   Please note that each function must implement a security mechanism to restrict access to vehicle properties to the authorised `AID_VEHICLE_NETWORK` process.
```

Comment:
I think this should use a firmer language, e.g.:
```
Please note that you must implement your own access control to vehicle properties to ensure secure access...
```

Or something along those lines. Wdyt?